volatility3/framework/plugins/windows/pedump.py

```diff
@@ -0,0 +1,270 @@
+            requirements.IntRequirement(
+                name="base",
+                description="Base address to reconstruct a PE file",
+                optional=False,
+            ),
+            requirements.BooleanRequirement(
+                name="kernel_module",
+                description="Extract from kernel address space.",
+                default=False,
+                optional=True,
+            ),
+        ]
+
+    @classmethod
+    def dump_pe(
+        cls,
+        context: interfaces.context.ContextInterface,
+        pe_table_name: str,
+        layer_name: str,
+        open_method: Type[interfaces.plugins.FileHandlerInterface],
+        file_name: str,
+        base: int,
+    ) -> Optional[str]:
+        """
+        Returns the filename of the dump file or None
+        """
+        try:
+            file_handle = open_method(file_name)
+
+            dos_header = context.object(
+                pe_table_name + constants.BANG + "_IMAGE_DOS_HEADER",
+                offset=base,
+                layer_name=layer_name,
+            )
+
+            for offset, data in dos_header.reconstruct():
+                file_handle.seek(offset)
+                file_handle.write(data)
+        except (
+            IOError,
+            exceptions.VolatilityException,
+            OverflowError,
+            ValueError,
+        ) as excp:
+            vollog.debug(f"Unable to dump PE file at offset {base}: {excp}")
+            return None
+        finally:
+            file_handle.close()
+
+        return file_handle.preferred_filename
+
+    @classmethod
+    def dump_ldr_entry(
+        cls,
+        context: interfaces.context.ContextInterface,
+        pe_table_name: str,
+        ldr_entry: interfaces.objects.ObjectInterface,
+        open_method: Type[interfaces.plugins.FileHandlerInterface],
+        layer_name: str = None,
+        prefix: str = "",
+    ) -> Optional[str]:
+        """Extracts the PE file referenced an LDR_DATA_TABLE_ENTRY (DLL, kernel module) instance
+
+        Args:
+            context: the context to operate upon
+            pe_table_name: the name for the symbol table containing the PE format symbols
+            ldr_entry: the object representing the module
+            open_method: class for constructing output files
+            layer_name: the layer that the DLL lives within
+            prefix: optional string to prepend to filename
+        Returns:
+            The output file name or None in the case of failure
+        """
+        try:
+            name = ldr_entry.FullDllName.get_string()
+        except exceptions.InvalidAddressException:
+            name = "UnreadableDLLName"
+
+        if layer_name is None:
+            layer_name = ldr_entry.vol.layer_name
+
+        file_name = "{}{}.{:#x}.{:#x}.dmp".format(
+            prefix,
+            ntpath.basename(name),
+            ldr_entry.vol.offset,
+            ldr_entry.DllBase,
+        )
+
+        return PEDump.dump_pe(
```

Comment:
Wouldn't this just be `cls`?  Just seems a little out of place?

Reply:
nice catch. Fixed.